GillianCore/GIL_Parser/parser_state.ml

```diff
@@ -0,0 +1,7 @@
+let procs_with_no_paths = ref Containers.SS.empty
+
+let internal_file = ref false
```

Comment:
Are you sure you want to make this a global reference ?
`procs_with_no_paths` is ok because it's program-wide, but `internal_file` is proc-wide, it's gonna change very often. I would prefer if you create a state record type that you'd pass to every function (using a tagged parameter)
Actually in a better world it we'd have a compilation-state monad but I'm not gonna make you do that.

Changing this is optional though, but at least write a `FIXME` next to it explaining, because it will not be compatible with multi-core ocaml when it becomes usable